src/model_ode.cpp

```diff
@@ -89,29 +90,45 @@ void Model_ODE::fJDiag(const realtype t, AmiVector &JDiag,
 void Model_ODE::fdxdotdw(const realtype t, const N_Vector x) {
     dxdotdw.reset();
     auto x_pos = computeX_pos(x);
-    fdxdotdw(dxdotdw.data(), t, N_VGetArrayPointer(x_pos),
-             unscaledParameters.data(), fixedParameters.data(), h.data(),
-             w.data());
-    fdxdotdw_colptrs(dxdotdw.indexptrs());
-    fdxdotdw_rowvals(dxdotdw.indexvals());
+    if (nw > 0) {
+        fdxdotdw_colptrs(dxdotdw.indexptrs());
+        fdxdotdw_rowvals(dxdotdw.indexvals());
+        fdxdotdw(dxdotdw.data(), t, N_VGetArrayPointer(x_pos),
+                 unscaledParameters.data(), fixedParameters.data(), h.data(),
+                 w.data());
+    }
 }
 
 void Model_ODE::fdxdotdp(const realtype t, const N_Vector x) {
```

Comment:
how does this influence `NewtonSolver::computeNewtonSensis`? I feel like a couple changes will be necessary there.

Reply:
I does, and I didn't implement it yet. Two solutions:
1. Easy way, which hopefully does not hamper performance: always create a new NVector, initialize with zeros, update the few non-zero entries using sparse implementation and the use it like it is done now.
2. Harder way, very likely to speed up the Newton solver, maybe even substantially: Compute sensis somehow via the sparse dxdotdp matrix and the KLU factorization. No clue how to do, but may be really fast, if it works.